### ISSUE TITLE
discarded_notification_center_observer false positive

### ISSUE THREAD

**Issue by hartbit** [realm/SwiftLint#1398]
I found that the `discarded_notification_center_observer` rule warns incorrectly when storing the returned observer in an array:

```
notificationObservers.append(notificationCenter.addObserver(
    forName: .UIKeyboardWillShow, object: nil, queue: nil)
{ _ in
    // stuff...
})
```

**Comment by mzaks** [comment]
I have even more fun scenario :)
```
let tokens = names.map {
            self.notificationCenter.addObserver(
                forName: $0,
                object: object,
                queue: queue,
                using: block
            )
        }
```

**Comment by lilyball** [comment]
I just hit this as well. This is a very surprising false positive as it seems like a fairly common thing to do.

**Comment by lilyball** [comment]
Nitpick: The title says "false negative" but it should be "false positive"

**Comment by jpsim** [comment]
If anyone's looking to contribute this bugfix, this would make for an excellent first-time contributor bugfix.

**Comment by cndouglas** [comment]
This was fixed by #3031, which was included in version 0.39.0.

**Comment by CraigSiemens** [comment]
The array scenario was fixed but the second example where there is an implicit return is still present which is why this issue is still open.

**Comment by jrtibbetts** [comment]
I still see it in the array scenario under 0.39.2.

**Comment by jrtibbetts** [comment]
This is still an issue.

**Comment by jrtibbetts** [comment]
Why do I have to keep bumping this?

**Comment by jpsim** [comment]
Unfortunately there's no easy way to fix this rule to avoid these false positives, so I'm changing this rule to opt-in: https://github.com/realm/SwiftLint/pull/3537
